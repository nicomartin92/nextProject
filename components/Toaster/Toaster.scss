@import '../../styles/index.scss';

.toast {
    width: 350px;
    padding: 15px;
    position: fixed;
    overflow-y: scroll;
    top: 0;
    right: 0;
    z-index: map-get($z-index, toaster);
    transition: all .3s ease-in;
    background: $colorBlack;

    .App & {
      opacity: 0;
      transform: translateX(350px);
    }

    a {
      color: $colorWhite;
    }

    &.-show {
      opacity: 1;
      transform: translateX(0);
    }

    &.top-left {
      top: 10px;
      left: 10px;
    }
    
    &.top-right {
      top: 10px;
      right: 10px;  
    }
    
    &.bottom-left {
      bottom: 10px;
      left: 10px;
    }
    
    &.bottom-right {
      bottom: 10px;
      right: 10px;
    }

    &__wrapper {
      position: relative;
      width: 100%;
    }

    &__close {
      position: absolute;
      top: 3px;
      right: 3px;

      svg {
        fill: $colorWhite;
      }
    }

    &__header {
      margin-bottom: 10px;
    }

    &__content {
      display: flex;
      align-items: center;
    }

    &__image,
    &__text {
      flex: 0 0 50%;
      width: 50%;
    }

    &__image {
      img {
        width: 60px;
      }
    }
}