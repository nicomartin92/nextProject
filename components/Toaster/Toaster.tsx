import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

/* SVG */
import { ReactComponent as CrossIcon } from '../../assets/cross-icon.svg';

/* styles */
import './Toaster.scss';

type MyProps = {
    item: {
        url: string,
        succes: string,
        text: string,
        image: string
    }
};

type MyState = {
    showToaster: boolean,
    timer: number
};

class Toaster extends Component<MyProps, MyState> {
    constructor(props: any) {
        super(props);

        this.state = {
            showToaster: false,
            timer: 3000
        }

        /* this.positionMap = {
            1: 'top-left',
            2: 'top-right',
            3: 'bottom-right',
            4: 'bottom-left',
        } */
    }

    toastDisplay(value: boolean) {
        this.setState({
            showToaster: value
        });

        setTimeout(() => {
            this.setState({
                showToaster: false
            })
        }, this.state.timer);
    }

    render() {
        return (
            <div className={this.state.showToaster ? "toast -show" : "toast"}>
                <div className="toast__wrapper">
                    <NavLink to={`/Car/${this.props.item.url}`} >
                        <div className="toast__header">
                            {this.props.item.succes}
                        </div>
                        <div className="toast__content">
                            <div className="toast__text">{this.props.item.text}</div>
                            <div className="toast__image">
                                <img src={this.props.item.image} alt={this.props.item.text} />
                            </div>
                        </div>
                    </NavLink>

                    <button className="toast__close" onClick={() => { this.toastDisplay(false) }}>
                        <CrossIcon />
                    </button>
                </div>
            </div>
        )
    }
}

export default Toaster;
