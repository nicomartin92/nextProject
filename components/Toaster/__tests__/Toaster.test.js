import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Toaster from '../Toaster';

const items = {
    showToaster: true,
    url: '/',
    success: 'success',
    text: 'text',
    image: '/'
}

describe('<Toaster>', () => {
    const wrapper = shallow(<Toaster item={items} />);

    it('renders a toaster', () => {
        //expect(slider.find('div.slider__wrapper').length).toEqual(1);
        expect(wrapper).toMatchSnapshot();
    });

    it('state showToaster should be false', () => {
        expect(wrapper.state().showToaster).toEqual(false);
        expect(wrapper.find('.toast__close').length).toEqual(1);
    });

    it('should click on button and should be false', () => {
        const firstButton = wrapper.find('.toast__close');

        firstButton.simulate('click');
        wrapper.setState({
            showToaster: true
        });

        expect(wrapper.state().showToaster).toEqual(true);

        setTimeout(() => {
            expect(wrapper.state().showToaster).toEqual(true);
            wrapper.unmount();
            done();
        }, 3000);
    });
});