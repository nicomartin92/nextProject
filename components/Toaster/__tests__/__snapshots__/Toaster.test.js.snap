// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toaster> renders a toaster 1`] = `ShallowWrapper {}`;
